core/domain/user_services.py

```diff
@@ -938,14 +1015,61 @@ def create_new_user(gae_id, email):
         raise Exception(
             'User %s already exists for gae_id %s.'
             % (user_settings.user_id, gae_id))
-
     user_id = user_models.UserSettingsModel.get_new_id('')
     user_settings = UserSettings(
         user_id, gae_id, email, feconf.ROLE_ID_EXPLORATION_EDITOR,
         preferred_language_codes=[constants.DEFAULT_LANGUAGE_CODE])
-    _save_user_auth_details(UserAuthDetails(user_id, gae_id))
-    _save_user_settings(user_settings)
-    create_user_contributions(user_id, [], [])
+    user_auth_details = UserAuthDetails(user_id, gae_id)
+    transaction_services.run_in_transaction(
+        _save_new_user_details, user_settings, user_auth_details)
+    return user_settings
+
+
+def create_new_profile(
+        gae_id, email, profile_pin=None, profile_display_alias=None):
+    """Creates a new profile.
+
+    Args:
+        gae_id: str. The GAE ID of the parent user trying to create a new
+            profile.
+        email: str. The email of the parent user trying to create a
+            new profile.
+        profile_pin: str or None. The PIN to be set for the new profile.
+        profile_display_alias: str or None. Display alias to be set for the
+            new profile.
+
+    Returns:
+        UserSettings. The newly-created user settings domain object.
+
+    Raises:
+        Exception. If a user with the given display alias already exists with
+            the given parent_user_id.
+        Exception. If the given parent user does not have its PIN set.
+    """
+
+    # New profile creation is done by full (parent) user of the account.
+    parent_user_auth_details = get_auth_details_by_gae_id(gae_id, strict=True)
+    if parent_user_auth_details.pin is None:
+        raise Exception(
+            'Pin must be set for a full user before creating a profile.')
+    parent_user_id = parent_user_auth_details.user_id
+
+    if is_display_alias_unique_within_the_account(
+            parent_user_id, profile_display_alias):
+        raise Exception(
+            'Display alias %s already exists in the account.'
+            % (profile_display_alias)
+        )
+
+    user_id = user_models.UserSettingsModel.get_new_id('')
+    user_settings = UserSettings(
+        user_id, gae_id, email, feconf.ROLE_ID_LEARNER,
+        preferred_language_codes=[constants.DEFAULT_LANGUAGE_CODE])
+    user_settings.display_alias = profile_display_alias
+    user_auth_details = UserAuthDetails(
+        user_id, None, profile_pin, parent_user_id)
+    transaction_services.run_in_transaction(
+        _save_new_user_details, user_settings, user_auth_details)
```

Comment:
There is no user_contributions for profiles right? Given that you are calling that in _save_new_user_details, wont that be a problem?

Reply:
No I think it's a part of creator flow. 
Using the approach @seanlip suggested to keep an inner function - now I am not creating `user_contributions` for profiles.